test/validation/api/chksum/chksum.c

```diff
@@ -313,10 +313,87 @@ static void chksum_ones_complement_udp_long(void)
 	CU_ASSERT(res == UDP_LONG_CHKSUM);
 }
 
+static uint16_t chksum_rfc1071(const void *p, uint32_t len)
+{
+	uint32_t sum = 0;
+	const uint16_t *data = p;
+
+	while (len > 1) {
+		sum += *data++;
+		len -= 2;
+	}
+
+	/* Add left-over byte, if any */
+	if (len > 0) {
+		uint16_t left_over = 0;
+
+		*(uint8_t *)&left_over = *(const uint8_t *)data;
+		sum += left_over;
+	}
+
+	/* Fold 32-bit sum to 16 bits */
+	while (sum >> 16)
+		sum = (sum & 0xffff) + (sum >> 16);
+
+	return sum;
+}
+
+/*
+ * 64-bit KISS RNGs
+ * George Marsaglia
+ * https://www.thecodingforums.com/threads/64-bit-kiss-rngs.673657
+ */
+
+static unsigned long long x = 1234567890987654321ULL, c = 123456123456123456ULL,
+			  y = 362436362436362436ULL, z = 1066149217761810ULL, t;
+
+#define MWC (t = (x << 58) + c, c = (x >> 6), x += t, c += (x < t), x)
+#define XSH (y ^= (y << 13), y ^= (y >> 17), y ^= (y << 43))
+#define CNG (z = 6906969069LL * z + 1234567)
+#define KISS (MWC + XSH + CNG)
+
+/*
+ * Test relatively short pseudorandom data. This test is trying to maximize the
+ * number of checksums computed.
+ */
+static void chksum_ones_complement_pseudorandom(void)
+{
+	const int size = (1 << 15);
+	const int max_len = 64;
```

Comment:
This is rather short length for typical UDP/TCP checksums. If implementation has different code paths for e.g. short and long packets, the long packet code would be not tested. Maybe the test case could have couple of loops for different max_len values: e.g. 1M x 64B, 100k x 640B, 10k x 9000B

It's OK for this kind of validation test to run for some number of seconds.

Reply:
I ended up with a different solution, but this now tests a wide range of data lengths. Please check.